Update the Buttons page

ryankeairns opened the pull request:
## Summary

TL;DR - this page was verbose, showing many variants, some seldom used or contrary to guidance.

Buttons are being updated as part of Borealis. In parallel, we wrote UX Guidelines to better clarify button usages in our product UIs. This PR gets the house in order with a more narrow focus on the predominant, default usage. The revised guidelines will ultimately be added to the new EUI+ site once ready.

- The order/IA has been revised to reflect what is presumably the most used content and/or logical order (base → adjacent → pattern only)
- Every section has been refined - on both the legacy and EUI+ docs sites - though only a few examples are shown below.
- Removed **Buttons with icons** section (not to be confused with **Icon button**) as it is now covered in **Basic button** and should be used sparingly
- Labeled **Split button**, **Toggle button**, and **Button groups within forms** as patterns (i.e. added a badge)

### Before and after
_Basic button_
![CleanShot 2025-03-13 at 07 37 58@2x](https://github.com/user-attachments/assets/04da7ab5-9176-4cd8-a5c2-43a1e484af2c)

_Flush empty button_
![CleanShot 2025-03-13 at 14 14 18@2x](https://github.com/user-attachments/assets/3d2212fa-a44d-4b15-8877-783a7f68a984)

_Icon button_
![CleanShot 2025-03-13 at 14 15 57@2x](https://github.com/user-attachments/assets/2cfdc155-533b-4ebb-974f-1480679845e1)

_Checked dark mode and mobile_
<img width="320" src="https://github.com/user-attachments/assets/c72d8850-4efa-4e72-be2b-ca31791130d2" />

_Partial screenshot of EUI+ Buttons page_
![CleanShot 2025-03-13 at 14 35 27@2x](https://github.com/user-attachments/assets/f799444a-a2fa-4819-ab73-bd3586ce97c2)


## QA

Remove or strikethrough items that do not apply to your PR.

### General checklist

- Browser QA
    - [x] Checked in both **light and dark** modes
    - [x] Checked in **mobile**
    - [ ] Checked in **Chrome**, **Safari**, **Edge**, and **Firefox**
    - [ ] Checked for **accessibility** including keyboard-only and screenreader modes
- Docs site QA
    - [x] Added **[documentation](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/documenting)**
    - [ ] Props have proper **autodocs** (using `@default` if default values are missing) and **[playground toggles](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/documenting/playgrounds.md)**
    - [ ] Checked **[Code Sandbox](https://codesandbox.io/)** works for any docs examples
- Code quality checklist
    - [ ] Added or updated **[jest](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/testing/unit-testing.md) and [cypress](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/testing/cypress-testing.md) tests**
    - [ ] Updated **[visual regression tests](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/testing/visual-regression-testing.md)**
- Release checklist
    - [ ] A **[changelog](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/documenting/changelogs.md)** entry exists and is marked appropriately.
    - [ ] If applicable, added the **breaking change** issue label (and filled out the breaking change checklist)
- Designer checklist
  - [ ] If applicable, [file an issue](https://github.com/elastic/platform-ux-team/issues/new/choose) to update [EUI's Figma library](https://www.figma.com/community/file/964536385682658129) with any corresponding UI changes. _(This is an internal repo, if you are external to Elastic, ask a maintainer to submit this request)_

kibanamachine commented:
Preview staging links for this PR:
- Docs site: https://eui.elastic.co/pr_8440/
- Storybook: https://eui.elastic.co/pr_8440/storybook
- New docs: https://eui.elastic.co/pr_8440/new-docs

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3641)
* Commit: bf4de73b328772454f56959d54be00de0baf97d9

<!--buildkite-pr-comment-eui-pull-request-test-and-deploy
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3641","number":3641,"commit":"bf4de73b328772454f56959d54be00de0baf97d9"}],"number":3641}
buildkite-pr-comment-->

weronikaolejniczak commented on packages/website/docs/components/navigation/button/overview.mdx:
````diff
@@ -19,444 +19,829 @@ The most standard button component is **EuiButton** which comes in two styles an
 ```
 
-## Buttons with icons
+### Flush empthy button
````
**blocking:**

Just this one small typo 😄

weronikaolejniczak commented on packages/website/docs/components/navigation/button/overview.mdx:
```diff
@@ -19,444 +19,829 @@ The most standard button component is **EuiButton** which comes in two styles an
-Setting the `isLoading` prop to true will add the loading spinner or swap the existing icon for the loading spinner and set the button to disabled. It is good practice to also rename the button to "Loading…".
+Buttons within a button group will automatically display a default browser tooltip containing the button `label` text. This can be customized or unset via the `title` property in your `options` button configuration.
+
+To instead display an **EuiToolTip** around your button(s), pass the `toolTipContent` property. You can also use `toolTipProps` to customize tooltip placement, title, and any other prop that [**EuiToolTip**](/docs/display/tooltip) accepts.
```
**non-blocking:**

This link doesn't work, would be great to fix it here since we're touching the page 😄 

```suggestion
To instead display an **EuiToolTip** around your button(s), pass the `toolTipContent` property. You can also use `toolTipProps` to customize tooltip placement, title, and any other prop that [**EuiToolTip**](../../display/tooltip.mdx) accepts.
```

weronikaolejniczak commented on packages/website/docs/components/navigation/button/overview.mdx:
```diff
@@ -19,444 +19,829 @@ The most standard button component is **EuiButton** which comes in two styles an
 
 When using colors other than `primary`, be sure that either the words or an icon also represents the status. For instance, don't rely on color alone to represent dangerous actions but use words like "Delete" not "Confirm". The `text` and `accent` colors should be used sparingly as they can easily be confused with other states like disabled and danger.
 
+All button components accept an `iconType` which must be an acceptable [**EuiIcon**](/docs/display/icons) type. Multi-color icons like app icons will be converted to single color. Icons can be displayed on the opposite side by passing `iconSide="right"`.
```
**non-blocking:**

This link doesn't work, would be great to fix it here since we're touching the page 😄 

```suggestion
All button components accept an `iconType` which must be an acceptable [**EuiIcon**](../../display/icons/overview.mdx) type. Multi-color icons like app icons will be converted to single color. Icons can be displayed on the opposite side by passing `iconSide="right"`.
```

weronikaolejniczak commented on packages/website/docs/components/navigation/button/overview.mdx:
**blocking:**

Empty button and Button group are in separate pages. We need to update:
`packages/website/docs/components/navigation/button/button_empty.mdx`
and `packages/website/docs/components/navigation/button/button_group.mdx`. 

I think it's enough to cut and paste from this file 😄

ryankeairns replied:
Done! Also went ahead and added button_icon and button_basic. The latter change makes 'Button' show up as a child in the IA and leaves the top level 'Buttons' to the overview content only. In this way, all child items are either standalone component (EuiButton, EuiButtonIcon, etc.) or patterns or guidelines.

weronikaolejniczak replied:
@ryankeairns that's incredible, and this answers our doubt regarding information architecture, how some components are mentioned in the top-level page and others are extracted into subpages which creates an unintuitive navigation experience. This looks much cleaner! I'm noting this down for M2 as well ✍🏻

weronikaolejniczak commented on packages/website/docs/components/navigation/button/overview.mdx:
```diff
@@ -19,444 +19,829 @@ The most standard button component is **EuiButton** which comes in two styles an
+
+When using button groups within compressed forms, match the form elements by adding `buttonSize="compressed"`. Compressed groups should always be `fullWidth` so they line up nicely in their small container unless they are icon only.
+
+For a more detailed example of how to integrate with forms, see the ["Complex example"](/docs/forms/layouts/compressed-forms/#complex-example) on the compressed forms page.
```
**non-blocking:**

This link doesn't work, would be great to fix it here since we're touching the page 😄

```suggestion
For a more detailed example of how to integrate with forms, see the ["Complex example"](../../forms/form_layouts/compressed_forms.mdx#complex-example) on the compressed forms page.
```

weronikaolejniczak reviewed (changes requested):
Amazing update, Ryan 🎸 I love the new examples with more interactivity. I would think about it as a pattern in M2.

I noticed just 2 blockers, otherwise I have some non-blocking suggestions, feel free to ignore and we'll update later (though I admit, it'd be great to apply them the sooner we get the chance to, aka here).

ryankeairns commented:
Thanks @weronikaolejniczak !

I made those fixes and went a step further to break out basic button, icon button, and patterns pages from the overview page. In this way, we now have a more complete and accurate IA:

<img width="240" src="https://github.com/user-attachments/assets/1db73cd3-c300-45d9-af11-c51662564f10" />

The overview page is now more succinct...

![CleanShot 2025-03-18 at 11 59 45@2x](https://github.com/user-attachments/assets/2e22ba21-8951-4026-a947-26b6950e7d1a)

kibanamachine commented:
Preview staging links for this PR:
- Docs site: https://eui.elastic.co/pr_8440/
- Storybook: https://eui.elastic.co/pr_8440/storybook
- New docs: https://eui.elastic.co/pr_8440/new-docs

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3662)
* Commit: 176377121153e570e0303830b97f508deb61ddd6

### Failed CI Steps
* [:hammer: EUI pull request test](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3662#0195aa9d-6630-4349-93f6-ac5461353303)

### History
* :green_heart: [Build #3641](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3641) succeeded bf4de73b328772454f56959d54be00de0baf97d9

<!--buildkite-pr-comment-eui-pull-request-test-and-deploy
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3662","number":3662,"commit":"176377121153e570e0303830b97f508deb61ddd6"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3641","number":3641,"commit":"bf4de73b328772454f56959d54be00de0baf97d9"}],"number":3662}
buildkite-pr-comment-->

ryankeairns commented on packages/website/docs/components/navigation/button/button_group.mdx:
```diff
@@ -247,17 +273,69 @@ export default () => {
-For a more detailed example of how to integrate with forms, see the ["Complex example"](/docs/forms/compressed-forms#complex-example) on the compressed forms page.
+When using button groups within compressed forms, match the form elements by adding `buttonSize="compressed"`. Compressed groups should always be `fullWidth` so they line up nicely in their small container unless they are icon only.
+
+For a more detailed example of how to integrate with forms, see the ["Complex example"](../../forms/layouts/compressed-forms#complex-example) on the compressed forms page.
```
@weronikaolejniczak the `.mdx` extension did now work in my local environment nor did the underscores. I'm curious if these links now work for you?

weronikaolejniczak replied:
@ryankeairns I didn't check locally, let me verify

kibanamachine commented:
Preview staging links for this PR:
- Docs site: https://eui.elastic.co/pr_8440/
- Storybook: https://eui.elastic.co/pr_8440/storybook
- New docs: https://eui.elastic.co/pr_8440/new-docs

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3665)
* Commit: 74479cda7bba38174ddedb2706d1a9c322c2c145

### History
* :broken_heart: [Build #3662](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3662) failed 176377121153e570e0303830b97f508deb61ddd6
* :green_heart: [Build #3641](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3641) succeeded bf4de73b328772454f56959d54be00de0baf97d9

<!--buildkite-pr-comment-eui-pull-request-test-and-deploy
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3665","number":3665,"commit":"74479cda7bba38174ddedb2706d1a9c322c2c145"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3662","number":3662,"commit":"176377121153e570e0303830b97f508deb61ddd6"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3641","number":3641,"commit":"bf4de73b328772454f56959d54be00de0baf97d9"}],"number":3665}
buildkite-pr-comment-->

weronikaolejniczak commented on packages/website/docs/components/navigation/button/patterns.mdx:
```diff
@@ -0,0 +1,173 @@
+
+## Split button
+
+<Badge>Pattern</Badge>
```
**non-blocking suggestion:**

I think we can put the badge inline with the heading, it looks a bit better IMO. The badge and the heading are now connected, as opposed to the equal distance creating flat hierarchy. It's a nuance and I could be biased here. Let's discuss 😄 

![Screenshot 2025-03-19 at 12 52 58](https://github.com/user-attachments/assets/0ac3034d-5320-483e-8a09-bcd7724f3b2e)

![Screenshot 2025-03-19 at 12 52 11](https://github.com/user-attachments/assets/374ba630-d900-44c3-a313-e6f43f1c7fbf)

weronikaolejniczak reviewed (changes requested):
The IA changes you've made to the Button section are so good, it is much more intuitive now!

**Blocking:**

Various links do not work. Starting with the buttons in the overview:

https://github.com/user-attachments/assets/b793ee62-66f6-4a47-9582-c185a886667b

The reason the links don't work for you is because after moving the sections into subpages, they all need `../` now to get out of the navigation directory:

![Screenshot 2025-03-19 at 12 44 17](https://github.com/user-attachments/assets/459745b0-06c7-488e-8207-517bd18a66a7)

If you want, I can fix them and push a commit here 😄

ryankeairns commented:
@weronikaolejniczak thanks and doh! I'll give it another try :)

ryankeairns commented:
@weronikaolejniczak Alright, let's try again...

- Fixed those two links to be `./basic` which followed the pattern of the adjacent two buttons for empty and icon.
- Moved the badges - good call!
- And, somehow, I lost some changes to the button_icon along the way yesterday. Really, that just splits the example two two sections which seems easy to read / keep the `EuiButtonIcon` section about the component only.

Thanks for the reviews!

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3669)
* Commit: 1df45dc75ff3faec9c41173045d70406cc4b87b0

### Failed CI Steps
* [:hammer: EUI pull request test](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3669#0195aed4-14ad-443c-b03d-88e8c35b2ad7)

### History
* :green_heart: [Build #3665](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3665) succeeded 74479cda7bba38174ddedb2706d1a9c322c2c145
* :broken_heart: [Build #3662](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3662) failed 176377121153e570e0303830b97f508deb61ddd6
* :green_heart: [Build #3641](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3641) succeeded bf4de73b328772454f56959d54be00de0baf97d9

<!--buildkite-pr-comment-eui-pull-request-test-and-deploy
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3669","number":3669,"commit":"1df45dc75ff3faec9c41173045d70406cc4b87b0"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3665","number":3665,"commit":"74479cda7bba38174ddedb2706d1a9c322c2c145"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3662","number":3662,"commit":"176377121153e570e0303830b97f508deb61ddd6"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3641","number":3641,"commit":"bf4de73b328772454f56959d54be00de0baf97d9"}],"number":3669}
buildkite-pr-comment-->

kibanamachine commented:
Preview staging links for this PR:
- Docs site: https://eui.elastic.co/pr_8440/
- Storybook: https://eui.elastic.co/pr_8440/storybook
- New docs: https://eui.elastic.co/pr_8440/new-docs
